examples/vision/object_detection_using_vision_transformer.py

```diff
@@ -0,0 +1,499 @@
+"""
+Title: Object detection with vision transformers
+Author: Karan V. Dave
+Date created: 2022/03/27
+Last modified: 2022/03/27
+Description: A simple keras implementation for object detection task using vision transformers.
+"""
+
+"""
+## Introduction
+
+This is an implementation of the
+[Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929)
+model by Alexey Dosovitskiy et al. applied to the object detection task.
+The model is trained on
+[Caltech 101 dataset](http://www.vision.caltech.edu/datasets/),
+to detect an airplane in the given image. The article demonstrates
+that a pure transformer applied directly to sequences of the image
+patches can perform well on object detection tasks.
+
+ This example requires TensorFlow 2.4 or higher, and
+ [TensorFlow Addons](https://www.tensorflow.org/addons/overview),
+ TFA is imported to use `AdamW` optimizer. However, `keras.optimizers.AdamW`
+ is likely to be included in Keras 2.10. TFA can be installed using the following command:
+
+```
+ pip install -U tensorflow-addons
+```
+"""
+
+"""
+## Imports and setup
+"""
+
+import numpy as np
+import tensorflow as tf
```

Comment:
Also `from tensorflow import keras` and `from tensorflow.keras import layers` since these 2 are commonly used modules in this file.